src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -102,32 +80,31 @@ class ExecutionStateVariableChanger {
     T m_changer;
 };
 
-Value ByteCodeInterpreter::interpret(ExecutionState* state, ByteCodeBlock* byteCodeBlock, size_t programCounter, Value* registerFile)
+
+OpcodeTable::OpcodeTable()
 {
 #if defined(COMPILER_GCC) || defined(COMPILER_CLANG)
-    if (UNLIKELY(byteCodeBlock == nullptr)) {
-        goto FillOpcodeTableLbl;
-    }
+    // Dummy bytecode execution to initialize the OpcodeTable.
+    ExecutionState state(nullptr);
+    size_t dummyCode = reinterpret_cast<size_t>(FillOpcodeTableAsmLbl);
+    ByteCodeInterpreter::interpret(&state, nullptr, reinterpret_cast<size_t>(&dummyCode), nullptr);
 #endif
+}
 
-    ASSERT(byteCodeBlock != nullptr);
-    ASSERT(registerFile != nullptr);
-
+Value ByteCodeInterpreter::interpret(ExecutionState* state, ByteCodeBlock* byteCodeBlock, size_t programCounter, Value* registerFile)
+{
+    state->m_programCounter = &programCounter;
     {
-        ExecutionStateProgramCounterBinder binder(*state, &programCounter);
-        char* codeBuffer = byteCodeBlock->m_code.data();
-        programCounter = (size_t)(codeBuffer + programCounter);
-
 #if defined(COMPILER_GCC) || defined(COMPILER_CLANG)
 #define DEFINE_OPCODE(codeName) codeName##OpcodeLbl
 #define DEFINE_DEFAULT
-#define NEXT_INSTRUCTION() \
-    goto*(((ByteCode*)programCounter)->m_opcodeInAddress);
+#define NEXT_INSTRUCTION() goto NextInstruction;
```

Comment:
It looks like every `NEXT_INSTRUCTION` should go through one more `goto` operation than before.
So, it would take 2 `goto` operations for jumping to the next bytecode, right?
This consecutive `goto` operations might be optimized well by the compiler, but I'm wondering the performance effect.
The performance is improved significantly by removing one if statement at the start of each interpret operation?

Reply:
.... I'm wondering the performance effect.
->This change cause a lots of performance effect. sunspider 717ms->671ms
I think ` goto*(((ByteCode*)programCounter)->m_opcodeInAddress);` generates a few assembly instruction.
but `goto NextInstruction` needs only one instruction.
Smaller code can cause performance effect.(I think)

The performance is improved significantly by removing one if statement at the start of each interpret operation?
-> I hope this change cause a lots of performance effect but it is not :( (671ms->667ms)